bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -1406,7 +1407,7 @@ public void recoveryAddEntry(ByteBuf entry, WriteCallback cb, Object ctx, byte[]
             LedgerDescriptor handle = getLedgerForEntry(entry, masterKey);
             synchronized (handle) {
                 entrySize = entry.readableBytes();
-                addEntryInternal(handle, entry, cb, ctx);
+                addEntryInternal(handle, entry, LedgerType.FORCE_ON_JOURNAL, cb, ctx);
```

Comment:
We must not  play with leader type to instruct immediate sync. The caller must not change ledgertype just because the caller wanted different behavior. We can do an explicit sync call or have another write which can send optional sync/persist flag in addition to ledger type.

Reply:
@jvrao. This makes sense. Changing.
I will review the recovery work in further patches

Reply:
fixed as @sijie suggested. recovery must always force fsync on journal it does not matter LedgerType